retrace-server.spec.in

```diff
@@ -101,6 +101,36 @@ rm -f %{buildroot}%{_infodir}/dir
 %define retrace_gid_uid 174
 getent group retrace > /dev/null || groupadd -f -g %{retrace_gid_uid} --system retrace
 getent passwd retrace > /dev/null || useradd --system -g retrace -u %{retrace_gid_uid} -b %{_sharedstatedir} -m -s /sbin/nologin retrace
+if grep -qs '^[a-z_][a-z0-9_]\{0,30\}:[0-9]\+:[0-9]\+\s*$' /etc/subuid; then
+    if ! grep -q '^retrace:' /etc/subuid; then
+        SUBUID=$(grep '^[a-z_][a-z0-9_]\{0,30\}:[0-9]\+:[0-9]\+\s*$' /etc/subuid \
+                 | sed 's/^[^:]*://' \
+                 | sort -n \
+                 | tail -n 1)
+        SUBUID_CNT=$(echo "$SUBUID" | sed -e 's/.*://' -e 's/\s*$//')
+        SUBUID=$(echo "$SUBUID" | sed 's/:.*//')
+        RETRACE_SUBUID_ENTRY=retrace:$((SUBUID+SUBUID_CNT+1)):65536
+        echo $RETRACE_SUBUID_ENTRY >> /etc/subuid
+    fi
+else
+    RETRACE_SUBUID_ENTRY=retrace:100000:65536
+    echo $RETRACE_SUBUID_ENTRY >> /etc/subuid
+fi
+if grep -qs '^[a-z_][a-z0-9_]\{0,30\}:[0-9]\+:[0-9]\+\s*$' /etc/subgid; then
+    if ! grep -q '^retrace:' /etc/subgid; then
+        SUBGID=$(grep '^[a-z_][a-z0-9_]\{0,30\}:[0-9]\+:[0-9]\+\s*$' /etc/subgid \
+                 | sed 's/^[^:]*://' \
+                 | sort -n \
+                 | tail -n 1)
+        SUBGID_CNT=$(echo "$SUBGID" | sed -e 's/.*://' -e 's/\s*$//')
+        SUBGID=$(echo "$SUBGID" | sed 's/:.*//')
+        RETRACE_SUBGID_ENTRY=retrace:$((SUBGID+SUBGID_CNT+1)):65536
+        echo $RETRACE_SUBGID_ENTRY >> /etc/subgid
+    fi
+else
+    RETRACE_SUBGID_ENTRY=retrace:100000:65536
```

Comment:
Just as an FYI, /etc/login.defs can define the ranges to something else.

Reply:
I added parsing of /etc/login.defs in case those variables are defined.